### PR TITLE
For #18268, #18269 - [Saved cards] Display a list of Saved cards fetched from the credit card storage

### LINKED ISSUE #18268: [Saved cards] Add a Saved cards screen for credit cards
This issue will provide a basic layout of the "Saved cards" screen which will list all the credit cards from the storage along with a "Add credit card" button. We will not handle display the individual icons of the credit card providers in this issue.

This also depends on https://github.com/mozilla-mobile/fenix/issues/18265  and https://github.com/mozilla-mobile/android-components/issues/9804 to properly fetch all the credit cards and display an obfuscated credit card number, but for the sake of this issue, we will be focused on layout and visuals of dummy data.

Screen Shot 2021-03-02 at 3 39 09 PM

### PR DESCRIPTION
**Pull request by gabrielluong**
Fixes #18268, #18269

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Review by Mugurell — COMMENTED** [review]
This looks awesome!

**Comment by Mugurell** [comment]
This looks really nice but I saw that the current UX doesn't exactly maps the mockup posted in the ticket

| expected | actual |
| -- | -- | 
| image | image |


&nbsp;
Also saw that although the `CreditCard` instance is passed to the `CreditCardEditorFragment` that instance is not used to populate the card fields to actually "edit" an existing card.
Starting with the empty fields and pressing save at any time will create and store a new card, even one with all fields left empty.

Are there follow-up changes / tickets for this?

**Comment by gabrielluong** [comment]
Yes, we still need more work to get the icons and credit card validations in before we can determine what credit card issuer is for a given credit card that is done in https://github.com/mozilla-mobile/android-components/issues/9813. Icons are to be added in https://github.com/mozilla-mobile/fenix/issues/18271.

File: app/src/main/res/layout/component_credit_cards.xml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    android:layout_width="match_parent"
+    android:layout_height="match_parent">
+
+    <ProgressBar
+        android:id="@+id/progress_bar"
+        style="@style/Widget.AppCompat.ProgressBar.Horizontal"
+        android:indeterminate="true"
+        android:layout_width="match_parent"
+        android:layout_height="8dp"
+        android:translationY="-3dp"
```
[the remark this answers is not in the document]
Comment (gabrielluong) [reply app/src/main/res/layout/component_credit_cards.xml:18]: I don't have any insight to this. This seems to be what was the most common styling for all the ProgressBar I have seen in our code.

**Review by Mugurell — APPROVED** [review]
Thank you!
Looks like a nice skeleton that can be built on.

**Pull request merged**